pkg/gardenlet/controller/shoot/shoot_control_reconcile.go

```diff
@@ -105,6 +105,10 @@ func (c *Controller) runReconcileShootFlow(o *operation.Operation, operationType
 			Name: "Syncing shoot cluster information to seed",
 			Fn:   flow.TaskFn(botanist.SyncClusterResourceToSeed).RetryUntilTimeout(defaultInterval, defaultTimeout),
 		})
+		_ = g.Add(flow.Task{
+			Name: "Ensuring that ShootState exists",
+			Fn:   flow.TaskFn(botanist.EnsureShootStateExists).RetryUntilTimeout(defaultInterval, defaultTimeout),
```

Comment:
Don't we need a dependency to the `syncClusterResourceToSeed` task as `createShootStateSyncReconcileFunc` will try to fetch the Cluster resource from the Seed?

Reply:
The clusterResource is synced before that in [shoot_control.go: 330](https://github.com/gardener/gardener/blob/master/pkg/gardenlet/controller/shoot/shoot_control.go#L330). 
The function returned from `createShootStateSyncReconcileFunc` should only be executed if an extension resource already exists -> and that generally already requires the cluster resource to exist.

Reply:
Ah, ok. Thank you!